lib/forecast.py

```diff
@@ -20,9 +22,53 @@
 )  # , Matérn
 import pickle
 import warnings
+import yaml
+
+from lib.forecast_technique import (
+    GaussianProcessForecaster,
+    GaussianProcessRecursiveForecaster,
+)
+from lib.dimensionality_reduction import (
+    DimensionalityReductionPCA,
+    DimensionalityReductionKernelPCA,
+)
+
+
+sys.path.insert(0, "../")
+
+path_to_nemo_directory = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+path_to_nemo_directory = Path(path_to_nemo_directory)
 
 warnings.filterwarnings("ignore")
 
+Forecast_techniques = {
+    "GaussianProcessForecaster": GaussianProcessForecaster,
+    "GaussianProcessRecursiveForecaster": GaussianProcessRecursiveForecaster,
+}
+Dimensionality_reduction_techniques = {
+    "PCA": DimensionalityReductionPCA,
+    "KernelPCA": DimensionalityReductionKernelPCA,
+}
+
+
+with open(path_to_nemo_directory / "techniques_config.yaml", "r") as f:
+    config = yaml.safe_load(f)
+
+
+if config["DR_technique"]["name"] not in Dimensionality_reduction_techniques:
+    raise KeyError(
+        f"DR_technique {config['DR_technique']['name']} not found. Have you specified a valid dimensionality reduction technique in the config file?"
+    )
+else:
+    DR_technique = Dimensionality_reduction_techniques[config["DR_technique"]["name"]]
+
+if config["Forecast_technique"]["name"] not in Forecast_techniques:
+    raise KeyError(
+        f"Forecast_technique {config['Forecast_technique']['name']} not found. Have you specified a valid forecasting technique in the config file?"
+    )
+else:
+    Forecast_technique = Forecast_techniques[config["Forecast_technique"]["name"]]
+
```

Comment:
Would it be best to move all of this to somewhere else like a `main_forecast.py`?  See file comments.